book/listings/todo_app/1/todo_row/mod.rs

```diff
@@ -0,0 +1,81 @@
+mod imp;
+
+use crate::todo_object::TodoObject;
+use glib::{BindingFlags, Object};
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::{glib, pango};
+use pango::{AttrList, Attribute};
+
+glib::wrapper! {
+    pub struct TodoRow(ObjectSubclass<imp::TodoRow>)
+    @extends gtk::Box, gtk::Widget;
+}
+
+impl Default for TodoRow {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl TodoRow {
+    pub fn new() -> Self {
+        Object::new(&[]).expect("Failed to create `TodoRow`.")
+    }
+
+    pub fn bind_item(&self, item: &TodoObject) {
+        // Get state
+        let imp = imp::TodoRow::from_instance(self);
+
+        // Bind `todo_object.completed` to `todo_row.completed_button.active` and save binding
+        // The binding is bidirectional so changes on both sides will be mirrored by the other side
+        let completed_button_binding = item
+            .bind_property("completed", &imp.completed_button.get(), "active")
+            .flags(BindingFlags::SYNC_CREATE | BindingFlags::BIDIRECTIONAL)
+            .build()
+            .expect("Could not bind properties");
+        imp.bindings.borrow_mut().push(completed_button_binding);
+
+        // Bind `todo_object.content` to `todo_row.content_label.label` and save binding
+        // The binding is bidirectional so changes on both sides will be mirrored by the other side
+        let content_label_binding = item
+            .bind_property("content", &imp.content_label.get(), "label")
+            .flags(BindingFlags::SYNC_CREATE | BindingFlags::BIDIRECTIONAL)
+            .build()
+            .expect("Could not bind properties");
+        imp.bindings.borrow_mut().push(content_label_binding);
+
+        // Bind `todo_object.completed` to `todo_row.content_label.attributed` and save binding
+        // We transform the boolean `completed` so that whenever completed is true
+        // the content of the label will be strikethrough
+        let completed_label_binding = item
```

Comment:
Can't you just b bind the checkbox state here? so you don't need to store it in the bindings list.

Reply:
Yes, I could do that.
I don't mind either way 😊

Reply:
Yes that would be better as that binding doesn't depend on the item to be "unbind" later and would simplify the code a bit